bindings/python/tests/bindings/test_tokenizer.py

```diff
@@ -0,0 +1,221 @@
+from ..utils import data_dir, roberta_files
+
+from tokenizers import AddedToken, Tokenizer
+from tokenizers.models import Model, BPE
+from tokenizers.pre_tokenizers import ByteLevel
+from tokenizers.processors import RobertaProcessing
+from tokenizers.normalizers import Lowercase
+
+
+class TestAddedToken:
+    def test_instantiate_with_content_only(self):
+        added_token = AddedToken("<mask>")
+        assert type(added_token) == AddedToken
+        assert str(added_token) == "<mask>"
+        assert (
+            repr(added_token)
+            == 'AddedToken("<mask>", rstrip=False, lstrip=False, single_word=False)'
+        )
+        assert added_token.rstrip == False
+        assert added_token.lstrip == False
+        assert added_token.single_word == False
+
+    def test_can_set_rstrip(self):
+        added_token = AddedToken("<mask>", rstrip=True)
+        assert added_token.rstrip == True
+        assert added_token.lstrip == False
+        assert added_token.single_word == False
+
+    def test_can_set_lstrip(self):
+        added_token = AddedToken("<mask>", lstrip=True)
+        assert added_token.rstrip == False
+        assert added_token.lstrip == True
+        assert added_token.single_word == False
+
+    def test_can_set_single_world(self):
+        added_token = AddedToken("<mask>", single_word=True)
+        assert added_token.rstrip == False
+        assert added_token.lstrip == False
+        assert added_token.single_word == True
```

Comment:
Maybe you can use https://docs.python.org/3/library/unittest.html, it's pretty nice/canonical

Reply:
(you can use pytest as a test runner but define all the tests using unittest – we found it's the best combo)

Reply:
Would you have any information/reading material about this? What are the advantages of combining both instead of just using pytest?